mydemoproject/demoapp1/views/book.py

```diff
@@ -1,30 +1,54 @@
 from django.contrib import messages
+from django.db.models import Q
 from django.shortcuts import redirect
 from django.views.generic import (ListView, DeleteView,
                                   UpdateView, CreateView)
 from django.urls import reverse_lazy
 from demoapp1.models.book import Book
 from demoapp1.models.inventory import Inventory
 from django.contrib.auth.mixins import LoginRequiredMixin
+from django.contrib.auth.mixins import PermissionRequiredMixin
 
+from demoapp1.views.permissions_user import user_only
 
-# ListView for Books
-class BookListView(LoginRequiredMixin,ListView):
+
+class BaseBookListView(LoginRequiredMixin, ListView):
     model = Book
-    template_name = 'demoapp1/books.html'
     context_object_name = 'books'
+    paginate_by = 2
 
     def get_queryset(self):
-        return Book.objects.select_related('inventory')
+        queryset = Book.objects.select_related('inventory').order_by('book_id')
+        query = self.request.GET.get('q')
+        if query:
+            queryset = queryset.filter(
+                Q(book_name__icontains=query) | Q(author_name__icontains=query)
+            )
+        return queryset
+
+
+# ListView for Books
+class BookListView(PermissionRequiredMixin, BaseBookListView):
+    template_name = 'demoapp1/books.html'
+    permission_required = 'demoapp1.view_book_admin'
+
+
+class UserBookListView(PermissionRequiredMixin, BaseBookListView):
+    template_name = 'demoapp1/book_list_user.html'
+    permission_required = 'demoapp1.view_book_newuser'
+
+    def dispatch(self, request, *args, **kwargs):
+        return user_only(super().dispatch)(request, *args, **kwargs)
```

Comment:
There should be a single view for listing books.
Check which user is currently logged in. If admin, show actions and if user, don't.